### PR TITLE
Add missing type definitions for package `probe-image-size` deep imports.

### PR DESCRIPTION
**Pull request by ddembo**
# What's changed?

When directly deep importing the parsing functions from `probe-image-size/lib/parse_sync`, e.g. `png.js`, `ico.js`, etc, TypeScript does not infer narrow enough types for the argument or return types.

This PR adds type definitions for the following functions, and ensures return types that are more narrow than what TypeScript infers:
* `probe-image-size/lib/parse_sync/bmp.js`
* `probe-image-size/lib/parse_sync/ico.js`
* `probe-image-size/lib/parse_sync/png.js`
* `probe-image-size/stream.js`

Also added some test cases for the above, to `probe-image-size-tests.ts`.

Unfortunately, I cannot allocate any more time towards adding types & tests for the other parse functions and for now this will need to be done by other members of the community.

# Checklist

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

# If changing an existing definition:

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [bmp.js](https://github.com/nodeca/probe-image-size/blob/master/lib/parse_sync/bmp.js), [ico.js](https://github.com/nodeca/probe-image-size/blob/master/lib/parse_sync/ico.js), [png.js](https://github.com/nodeca/probe-image-size/blob/master/lib/parse_sync/png.js).
- [N/A] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

File: types/probe-image-size/index.d.ts [hunk 1]
```diff
@@ -1,4 +1,8 @@
 /// <reference types="node" />
+/// <reference path="./lib/parse_sync/bmp.js/index.d.ts" />
+/// <reference path="./lib/parse_sync/ico.js/index.d.ts" />
+/// <reference path="./lib/parse_sync/png.js/index.d.ts" />
+/// <reference path="./stream.js/index.d.ts" />
```
Comment (jakebailey) [inline types/probe-image-size/index.d.ts:5]: There shouldn't be any need to reference these from the root. An import should load them just fine.

Comment (jakebailey) [reply types/probe-image-size/index.d.ts:5]: (You should write a test for them)

**Review by jakebailey — CHANGES_REQUESTED** [review]

File: types/probe-image-size/index.d.ts (hunk 1 above)
Comment (ddembo) [reply types/probe-image-size/index.d.ts]: Thanks for the review @jakebailey, I've removed the references. Regarding writing tests, I did add tests for each added deep import in `./probe-image-size-tests.ts`; were there any test cases that you felt I was missing? Or is there a better pattern for testing deep imports?

Comment (jakebailey) [reply types/probe-image-size/index.d.ts]: Tests should be written how a user is expected to use the library. If that means the deep imports, then that would be the thing to do.

File: types/probe-image-size/tsconfig.json [hunk 2]
```diff
@@ -14,6 +14,10 @@
     },
     "files": [
         "index.d.ts",
-        "probe-image-size-tests.ts"
+        "probe-image-size-tests.ts",
+        "lib/parse_sync/bmp.js/index.d.ts",
+        "lib/parse_sync/ico.js/index.d.ts",
+        "lib/parse_sync/png.js/index.d.ts",
```
Comment (jakebailey) [inline types/probe-image-size/tsconfig.json:20]: Why aren't these `lib/parse_sync/bmp.d.ts`, `ico.d.ts`, etc?

Comment (ddembo) [reply types/probe-image-size/tsconfig.json]: Thank you for the suggestion, I've improved the file structure

File: types/probe-image-size/index.d.ts (hunk 1 above)
Comment (ddembo) [reply types/probe-image-size/index.d.ts]: I believe that is the case already, e.g.:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/144092d239566f8dd1f8dd3c4953b3ab74356cda/types/probe-image-size/probe-image-size-tests.ts#L33-L52

**Comment by jakebailey** [comment]
One last question; these entrypoints are not documented by the package at all. Are they intended to be imported like this?

**Comment by ddembo** [comment]
I can only speculate about what the creator intended or not, but the first key feature is `small size, no heavy dependencies` and I think that deep-importing individual functions is a manifestation that philosophy. Also further down in the API section, they further seem to encourage deep importing:

> You can access/browserify stream.js / http.js / sync.js directly.

**Review by jakebailey — APPROVED** [review]

**Comment by ddembo** [comment]
Ready to merge

**Pull request merged**